packages/web/src/components/search/DataSearch.js

```diff
@@ -343,22 +353,44 @@ class DataSearch extends Component {
 
 	handleTextChange = debounce((value) => {
 		if (this.props.autosuggest) {
-			this.updateQuery(this.internalComponent, value, this.props);
+			this.updateDefaultQuery(value, this.props);
 		} else {
-			this.updateQuery(this.props.componentId, value, this.props);
+			this.updateQuery(value, this.props);
 		}
 	}, this.props.debounce);
 
-	updateQuery = (componentId, value, props) => {
+	updateDefaultQuery = (value, props) => {
+		const { defaultQuery } = props;
+		let defaultQueryOptions;
+		let query = DataSearch.defaultQuery(value, props);
+		if (defaultQuery) {
+			const defaultQueryTobeSet = defaultQuery(value, props) || {};
+			if (defaultQueryTobeSet.query) {
+				({ query } = defaultQueryTobeSet);
+			}
+			defaultQueryOptions = getOptionsFromQuery(defaultQueryTobeSet);
+		}
+		props.setQueryOptions(this.internalComponent, {
+			...this.queryOptions,
+			...defaultQueryOptions,
+		});
+		props.updateQuery({
+			componentId: this.internalComponent,
+			query,
+			value,
+			componentType: 'DATASEARCH',
```

Comment:
Can we make ENUM for all component type we have and use it? WDYT?

Reply:
> Can we make ENUM for all component type we have and use it? WDYT?

At that time it may need some extra work since they have been defined in `reactivecore` so we have to export them and use it here.
I think it's ok to just use strings, we can do that code refactoring later at some time.